### ISSUE TITLE
Deployment issue with Route Handler

### ISSUE THREAD

**Issue by ryanhay** [nextauthjs/next-auth#7229]
### Environment

  System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 16.40 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/Library/Caches/fnm_multishells/2264_1681008406169/bin/node
    npm: 9.5.0 - ~/Library/Caches/fnm_multishells/2264_1681008406169/bin/npm
  Browsers:
    Brave Browser: 112.1.50.114
    Chrome: 112.0.5615.49
    Firefox Developer Edition: 112.0
    Safari: 16.4

### Reproduction URL

private repo

### Describe the issue

When initialising NextAuth with the 'app dir' and the env variable _AUTH_TRUST_HOST_ set the function **detectHost** always returns the default value 'http://localhost:3000'.

It seems the **NextAuthRouteHandler** uses the same method as the **NextAuthApiHandler** to get the host header.

Currently uses: `req.headers["x-forwarded-host"]` instead of `req.headers.get('x-forwarded-host')`.

### How to reproduce

Install 4.22 and initialise in the app dir with either _AUTH_TRUST_HOST_ or _VERCEL_ env vars set.

### Expected behavior

When the env variable _AUTH_TRUST_HOST_ is set the value of _x-forwarded-host_ should be used instead of http://localhost:3000.

**Comment by bugplanter** [comment]
Possibly related to [#7220](https://github.com/nextauthjs/next-auth/pull/7220)

**Comment by hellodamien** [comment]
Yes, we have the same issue #7220 

Maybe a fix can be : https://github.com/hellodamien/next-auth/commit/1870a38b6d5e2c3a229cfd4283958c8396264b4f

**Comment by balazsorban44** [comment]
I could identify this as an upstream Next.js bug, we will look into it there! I'll leave this open to be able to track the issue, but there is no code changes that would be needed in Auth.js once it's fixed in `next`

**Comment by balazsorban44** [comment]
Update, I pushed a temporary fix in `next-auth@4.22.1` to unblock people or have to apply a workaround. We will follow the upstream releases and will handle changes internally, no change is needed on the user side.

**Comment by enyelsequeira** [comment]
Hello, I am wondering if this is also related to this issue, I am deploying a next app with `route handlers` and although my configuration is working on local on deployed version it doesn't  
![image](https://user-images.githubusercontent.com/50872387/232878599-5d2339f4-fe86-497b-adce-86460b63907f.png)
![image](https://user-images.githubusercontent.com/50872387/232878733-5314a864-77f8-4fad-855f-5546cb13b5d1.png)

![image](https://user-images.githubusercontent.com/50872387/232878409-d9d8d8af-4785-4834-b920-6d516449b5ab.png)

Using github immediately get this 
http://localhost:3000/api/auth/error?error=Callback

and then on google 
http://localhost:3000/api/auth/callback/google?state=a7Q6uP-6yY-Ox81WSzpDfUAZY0ZqWzZ9oLtTe_1YnUE&code=4%2F0AVHEtk5rbL4ZT8e9a8jYklrdjy8aNE7A8MRmyArX33ku5psoTRCRlurHEPjos5TlhevE_g&scope=email+profile+openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile&authuser=0&prompt=consent

**Comment by balazsorban44** [comment]
Please open a separate issue with a reproduction. Preferably don't just give a huge repro, please try to cut down the code, and make sure you are trying out `next-auth@4.22.1`

**Comment by enyelsequeira** [comment]
> Please open a separate issue with a reproduction. Preferably don't just give a huge repository, please try to cut down the code that is easy to investigate, and make sure you are trying out `next-auth@4.22.1`. 🙏
> 
> Note, on Vercel you don't need to set `NEXTAUTH_URL` as [documented](https://next-auth.js.org/deployment#vercel).


I updated to latest version and this seems to resolve the issue with next 13 app dir. Thank you  🙏🏽

**Comment by ryanhay** [comment]
4.22.1 works for me too, thank you!

**Comment by ROHAN13498** [comment]
https://issue-tracker-kappa-one.vercel.app/ I am using 4.22.1 but I am getting the same error after I click login .are there ay other ways to solve this?

![image](https://github.com/nextauthjs/next-auth/assets/95855114/4911de84-2d3c-48a5-aa2e-4134e0d5c0eb)
